ledger/participant-state/kvutils/app/src/test/scala/com/daml/ledger/participant/state/kvutils/app/ConfigSpec.scala

```diff
@@ -3,31 +3,66 @@
 
 package com.daml.ledger.participant.state.kvutils.app
 
+import com.daml.ledger.participant.state.v1.ParticipantId
 import org.scalatest.{FlatSpec, Matchers, OptionValues}
 import scopt.OptionParser
 
 final class ConfigSpec extends FlatSpec with Matchers with OptionValues {
 
-  private val DumpIndexMetadataCommand = "dump-index-metadata"
-  private val ConfigParser: Seq[String] => Option[Config[Unit]] =
-    Config.parse("Test", (_: OptionParser[Config[Unit]]) => (), (), _)
+  private val dumpIndexMetadataCommand = "dump-index-metadata"
+  private val participantOption = "--participant"
+  private val participantId: ParticipantId = ParticipantId.assertFromString("dummy-participant")
+  private val fixedParticipantSubOptions = s"participant-id=$participantId,port=123"
+  private val jdbcUrlSubOption = "server-jdbc-url"
+  private val jdbcUrlEnvSubOption = "server-jdbc-url-env"
+  private def configParser(
+      parameters: Seq[String],
+      getEnvVar: String => Option[String] = (_ => None)): Option[Config[Unit]] =
+    Config.parse("Test", (_: OptionParser[Config[Unit]]) => (), (), parameters, getEnvVar)
 
   behavior of "Runner"
 
   it should "fail if a participant is not provided in run mode" in {
-    ConfigParser(Seq.empty) shouldEqual None
+    configParser(Seq.empty) shouldEqual None
   }
 
   it should "fail if a participant is not provided when dumping the index metadata" in {
-    ConfigParser(Seq(DumpIndexMetadataCommand)) shouldEqual None
+    configParser(Seq(dumpIndexMetadataCommand)) shouldEqual None
   }
 
   it should "succeed if a participant is provided when dumping the index metadata" in {
-    ConfigParser(Seq(DumpIndexMetadataCommand, "some-jdbc-url"))
+    configParser(Seq(dumpIndexMetadataCommand, "some-jdbc-url"))
   }
 
   it should "succeed if more than one participant is provided when dumping the index metadata" in {
-    ConfigParser(Seq(DumpIndexMetadataCommand, "some-jdbc-url", "some-other-jdbc-url"))
+    configParser(Seq(dumpIndexMetadataCommand, "some-jdbc-url", "some-other-jdbc-url"))
+  }
+
+  it should "get the jdbc string from the command line argument when provided" in {
+    val jdbcFromCli = "command-line-jdbc"
+    val config = configParser(
+      Seq(participantOption, s"$fixedParticipantSubOptions,$jdbcUrlSubOption=$jdbcFromCli"))
```

Comment:
To improve test readability, I'd suggest putting the relevant config key first (i.e., from this test case's perspective).